server/service/src/document/mod.rs

```diff
@@ -1,28 +1,24 @@
+use chrono::{DateTime, Utc};
 use repository::RepositoryError;
 
 use crate::service_provider::{ServiceContext, ServiceProvider};
 
-use self::raw_document::RawDocument;
-
 pub mod document_registry;
 pub mod document_service;
 pub mod form_schema_service;
 pub mod raw_document;
 
-/// Checks that there is no document with a later timestamp
+/// Checks if the doc is the latest in the DB
 pub(crate) fn is_latest_doc(
     ctx: &ServiceContext,
     service_provider: &ServiceProvider,
-    doc: &RawDocument,
+    doc_name: &str,
+    doc_timestamp: DateTime<Utc>,
 ) -> Result<bool, RepositoryError> {
-    let latest_existing = service_provider
+    let Some(latest_existing) = service_provider
```

Comment:
Great the `let Some` works quite nice imo.

Reply:
Although, just realised this looks like a simple `map`, sorry only suggesting this because it look like it was refactored due to stylistic preference
```rust
    Ok(service_provider
        .document_service
        .document(ctx, doc_name, None)?
        .map_or(false, |r| r.datetime <= doc_timestamp))
```

Reply:
yes would work as well (`.map_or(true,...)` though).

Site note, find `map_or` relative uncommon so couldn't understand the code without looking at the docs... had this as a general observation, functional style code is sometimes simply harder because you need to know the used lib much better...